airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/http_status_error_handler.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import logging
+from typing import Mapping, Optional, Type, Union
+
+import requests
+from airbyte_cdk.models import FailureType
+from requests import RequestException
+
+from .error_handler import ErrorHandler
+from .response_models import ErrorResolution, ResponseAction
+
+
+class HttpStatusErrorHandler(ErrorHandler):
+
+    _DEFAULT_ERROR_MAPPING: Mapping[Union[int, str, Type[Exception]], ErrorResolution] = {
+        RequestException: ErrorResolution(
+            response_action=ResponseAction.RETRY,
+            failure_type=FailureType.transient_error,
+            error_message="An exception occurred when making the request.",
+        ),
+        400: ErrorResolution(
+            response_action=ResponseAction.FAIL,
+            failure_type=FailureType.config_error,
```

Comment:
Why is the failure_type for a 400 a config error? It can be, but I think it's been a pretty common case that we the developers have written code that misconfigured the outbound request that triggers a 400 error. And there are some sources like FB that return a 400 errors for everything.

Unless anyone is opposed, I think this should remain a system error as the default behavior so we don't lose visibility on legitimate errors since I'm not sure we can generalize this always a config error.

Reply:
@pnilan let me know what you think of this, otherwise good on my end

Reply:
I think that 400 by default should be a config error and the case of FB returning 400 for everything is a unique, one-off case where we'd probably want to override the mapping anyway to be more granular.

Reply:
I still have some concerns about this. Although a 400 indicates an issue with the request we make to the API, it is not always specifically a problem with the user's configuration. For example, if we introduce a bug in the code that adds an extra invalid param to the request (not something a customer did), there is no action on their part to fix the issue. And the what is worse is we can lose visibility of the problem because the platform doesn't surface config errors to Sentry.

I did a cursory search of our OC issue board and there are a lot of 400 errors across a lot of sources that are valid issues. If we were to switch this to config, we would miss a lot of these types of issues and regressions to a connector

The way I see the delineation is the most simplistic way is `config_error` is something that is user actionable they can fix.

Reply:
After discussion. I agree it makes more sense to set as a system error default and put the onus on the connector developer to remap if needed.